apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx

```diff
@@ -0,0 +1,184 @@
+import { useIntl } from '@core/next/intl'
+import { find, get, differenceBy, uniqBy } from 'lodash'
+import { UserNameField } from '@condo/domains/user/components/UserNameField'
+import { Col, Form, Row, Select, Typography } from 'antd'
+import { AutoAssignerByDivisions } from './AutoAssignerByDivisions'
+import { LabelWithInfo } from '@condo/domains/common/components/LabelWithInfo'
+import { GraphQlSearchInput } from '@condo/domains/common/components/GraphQlSearchInput'
+import { searchEmployeeUser } from '../../utils/clientSchema/search'
+import React, { useState } from 'react'
+
+const TicketAssignments = ({ validations, organizationId, propertyId, disableUserInteraction, autoAssign, categoryClassifier, form }) => {
+    const intl = useIntl()
+    const TicketAssignmentTitle = intl.formatMessage({ id: 'TicketAssignment' })
+    const ExecutorLabel = intl.formatMessage({ id: 'field.Executor' })
+    const ResponsibleLabel = intl.formatMessage({ id: 'field.Responsible' })
+    const ExecutorExtra = intl.formatMessage({ id: 'field.Executor.description' })
+    const ResponsibleExtra = intl.formatMessage({ id: 'field.Responsible.description' })
+    const ExecutorsOnThisDivisionLabel = intl.formatMessage({ id: 'ticket.assignments.executor.OnThisDivision' })
+    const ExecutorsOnOtherDivisionsLabel = intl.formatMessage({ id: 'ticket.assignments.executor.OnOtherDivisions' })
+    const OtherExecutors = intl.formatMessage({ id: 'ticket.assignments.executor.Other' })
+
+    const [divisions, setDivisions] = useState([])
+
+    const formatUserFieldLabel = ({ text, value, data: employee }) => {
+        if (!employee) {
+            return null
+        }
+        const matchedSpecialization = find(employee.specializations, { id: categoryClassifier })
+        return (
+            <UserNameField user={{ name: text, id: value }}>
+                {({ name, postfix }) => (
+                    <>
+                        <Typography.Text>
+                            {name} {postfix}
+                        </Typography.Text>
+                        {matchedSpecialization && (
+                            <Typography.Text type="secondary">
+                                {`(${matchedSpecialization.name.toLowerCase()})`}
+                            </Typography.Text>
+                        )}
+                    </>
+                )}
+            </UserNameField>
+        )
+    }
+
+    const getTechniciansFrom = (division) => (
+        division.executors.filter(({ specializations }) => (
+            specializations.some(({ id }) => id === categoryClassifier)
+        ))
+    )
+
+    const convertToOption = (employee) => ({
+        text: employee.name,
+        value: employee.user.id,
+        data: employee,
+    })
+
+    /**
+     * Employees are grouped by following rules:
```

Comment:
Should it be client side only logic? Maybe we should move this to serverce?

Reply:
This use case is only presented in CRM desktop app for Organisations yet in ticket form and this is a semi-automatic feature, that just pre-selects executor and responsible, allowing user to change selection while creating a ticket.

Reply:
All feature with divisions is introduced only for ability to preselect assignee and executor;) I'm not sure this is client side only logic.
Guess we need to introduce dedicated service for this.

Reply:
I agree, I will extract grouped suggestions of executors into custom GraphQL query.
Created separate task DOMA-1212 for it.
Thank You for this point.